src/Ledger/Certs/Properties/PoV.lagda

```diff
@@ -0,0 +1,84 @@
+\begin{code}[hide]
+{-# OPTIONS --safe #-}
+
+open import Ledger.Types.GovStructure
+
+module Ledger.Certs.Properties.PoV (gs : _) (open GovStructure gs) where
+
+\end{code}
+% If the module name changes, change the following macro to match!
+\newcommand{\CertsPoV}{Certs/Properties/PoV}
+
+\begin{code}[hide]
+open import Ledger.Certs gs
+open import Ledger.Certs.Properties.PoVLemmas gs
+open import Ledger.GovernanceActions gs hiding (yes; no)
+open import Ledger.Prelude
+
+open import Axiom.Set.Properties th
+
+open import Algebra using (CommutativeMonoid)
+open import Data.Maybe.Properties
+open import Data.Nat.Properties using (+-0-monoid; +-0-commutativeMonoid; +-identityʳ; +-identityˡ)
+open import Relation.Binary using (IsEquivalence)
+open import Relation.Nullary.Decidable
+open import Tactic.ReduceDec
+
+open Computational ⦃...⦄
+
+open import Tactic.GenError using (genErrors)
+
+open CertState
+
+private variable
+  dCert : DCert
+  l : List DCert
+  A A' B : Type
+instance
+  _ = +-0-monoid
+
+module _  ( indexedSumᵛ'-∪' :  {A : Type} ⦃ _ : DecEq A ⦄ (m m' : A ⇀ Coin)
+                              → disjoint (dom m) (dom m')
+                              → getCoin (m ∪ˡ m') ≡ getCoin m + getCoin m' )
+    -- TODO: prove some or all of the following assumptions, used in roof of `CERTBASE-pov`.
+    ( sumConstZero'    :  {A : Type} ⦃ _ : DecEq A ⦄ {X : ℙ A} → getCoin (constMap X 0) ≡ 0 )
+    ( res-decomp'      :  {A : Type} ⦃ _ : DecEq A ⦄ (m m' : A ⇀ Coin)
+                         → (m ∪ˡ m')ˢ ≡ᵉ (m ∪ˡ (m' ∣ dom (m ˢ) ᶜ))ˢ )
+    ( getCoin-cong'    :  {A : Type} ⦃ _ : DecEq A ⦄ (s : A ⇀ Coin) (s' : ℙ (A × Coin)) → s ˢ ≡ᵉ s'
+                         → indexedSum' proj₂ (s ˢ) ≡ indexedSum' proj₂ s' )
+    ( ≡ᵉ-getCoinˢ'     :  {A A' : Type} ⦃ _ : DecEq A ⦄ ⦃ _ : DecEq A' ⦄ (s : ℙ (A × Coin)) {f : A → A'}
+                         → InjectiveOn (dom s) f → getCoin (mapˢ (map₁ f) s) ≡ getCoin s )
+    where
+    open CERTSpov indexedSumᵛ'-∪' sumConstZero' res-decomp' getCoin-cong' ≡ᵉ-getCoinˢ'     
+\end{code}
+
+\begin{theorem}[%
+  \LedgerMod{\CertsPoV.lagda}{\AgdaModule{\CertsPoV{}}}:
+  \CERTS{} rule preserves value%
+  ]\label{thm:CERTS-PoV}
+  \begin{itemize}
+    \item \textit{Informally}.
+      Let \ab{l} be a list of \DCert{}s, and let \ab{s₁}, \ab{sₙ} be \CertState{}s
+      such that \ab{s₁}~\AgdaDatatype{⇀⦇}~\ab{l}~\AgdaDatatype{,CERTS⦈}~\ab{sₙ}.
+      Then, the value of \ab{s₁} is equal to the value of \ab{sₙ} plus
+      the value of the withdrawals in \ab{Γ}.
+    \item \textit{Formally}.
+\begin{code}
+    CERTS-pov :  {Γ : CertEnv} {s₁ sₙ  : CertState}
+      → ∀[ a ∈ dom (CertEnv.wdrls Γ) ] NetworkIdOf a ≡ NetworkId
```

Comment:
This assumption is not mentioned in the prose. Why is it needed?

Reply:
Not sure. 🤷🏼